src/System Application/App/Performance Profiler/src/PerformanceProfilerInstaller.Codeunit.al

```diff
@@ -0,0 +1,66 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Tooling;
+
+using System.Environment.Configuration;
+using System.DataAdministration;
+using System.PerformanceProfile;
+using System.Reflection;
+using System.Upgrade;
+
+codeunit 1933 "Performance Profiler Installer"
+{
+    Subtype = Install;
+    Access = Internal;
+    InherentPermissions = X;
+    InherentEntitlements = X;
+    Permissions = tabledata Field = r;
+
+    trigger OnInstallAppPerCompany()
+    begin
+        AddRetentionPolicyAllowedTables(false);
+    end;
+
+    procedure AddRetentionPolicyAllowedTables(ForceUpdate: Boolean)
```

Comment:
you want to call the same code from the upgrade. The apps are probably not installed in SaaS, I'm not sure if we are raising this trigger.

Reply:
We are following the existing pattern, like in EmailInstaller.Codeunit.al. This is the system application, so it will be installed in SaaS because base app depends on it.